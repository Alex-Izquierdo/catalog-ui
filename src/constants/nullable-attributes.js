export const PORTFOLIO_ITEM_NULLABLE = [
  'display_name', 'description', 'long_description', 'distributor', 'documentation_url', 'support_url', 'workflow_ref'
];
