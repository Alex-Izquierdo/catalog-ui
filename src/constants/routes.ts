export const ADD_PORTFOLIO_ROUTE = '/portfolios/add-portfolio';
export const ADD_PRODUCTS_ROUTE = '/portfolios/add-products';
export const EDIT_ORDER_PROCESS_ROUTE = '/portfolios/edit-order-process';
export const EDIT_PORTFOLIO_ROUTE = '/portfolios/edit';
export const EDIT_PORTFOLIO_WORKFLOW_ROUTE = '/portfolios/edit-workflow';
export const EDIT_SURVEY_ROUTE = '/portfolios/portfolio-item/edit-survey';
export const NESTED_EDIT_ORDER_PROCESS_ROUTE =
  '/portfolios/portfolio/edit-order-process';
export const NESTED_EDIT_PORTFOLIO_ROUTE =
  '/portfolios/portfolio/edit-portfolio';
export const NESTED_REMOVE_PORTFOLIO_ROUTE =
  '/portfolios/portfolio/remove-portfolio';
export const NESTED_SHARE_PORTFOLIO_ROUTE =
  '/portfolios/portfolio/share-portfolio';
export const NESTED_WORKFLOW_PORTFOLIO_ROUTE =
  '/portfolios/portfolio/edit-workflow';
export const ORDER_ROUTE = '/orders/order';
export const ORDERS_ROUTE = '/orders';
export const PLATFORM_INVENTORIES_ROUTE = '/platforms/platform-inventories';
export const PLATFORM_ROUTE = '/platforms/platform';
export const PLATFORM_SERVICE_OFFERINGS_ROUTE = '/platforms/service-offerings';
export const PLATFORM_TEMPLATES_ROUTE = '/platforms/platform-templates';
export const PLATFORM_DETAILS_ROUTE = '/platforms/platform-details';
export const PLATFORMS_ROUTE = '/platforms';
export const PORTFOLIO_ROUTE = '/portfolios/portfolio';
export const PORTFOLIO_ITEM_ROUTE = '/portfolios/portfolio-item';
export const PORTFOLIO_ITEM_ORDER_ROUTE = '/portfolios/portfolio-item/order';
export const PORTFOLIO_ITEM_ROUTE_EDIT = '/portfolios/portfolio-item/edit';
export const PORTFOLIO_ITEM_EDIT_ORDER_PROCESS_ROUTE =
  '/portfolios/portfolio-item/edit-order-process';
export const PORTFOLIOS_ROUTE = '/portfolios';
export const REMOVE_PORTFOLIO_ROUTE = '/portfolios/remove';
export const SHARE_PORTFOLIO_ROUTE = '/portfolios/share';
export const WORKFLOW_PORTFOLIO_ROUTE = '/portfolios/edit-workflow';
export const ORDER_PROCESSES_ROUTE = '/order-processes';
export const ADD_ORDER_PROCESS_ROUTE = '/order-processes/add-order-process';
export const REMOVE_ORDER_PROCESS_ROUTE = '/order-processes/remove';
export const UPDATE_ORDER_PROCESS_ROUTE = '/order-processes/update';
