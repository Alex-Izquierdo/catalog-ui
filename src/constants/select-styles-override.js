const style = {
  container: provided => ({
    ...provided,
    '&:focus': {
      outline: 'none'
    }
  }),
  control: (provided, { isFocused }) => ({
    ...provided,
    borderRadius: 0,
    borderColor: '#EDEDED',
    backgroundColor: '#FFFFFF',
    borderBottomColor: isFocused ? '#007bba' : '#282d33',
    cursor: 'pointer',
    boxShadow: 'none',
    minHeight: 36,
    width: 256,
    '&:hover': {
      borderBottomColor: '#007bba'
    }
  }),
  indicatorSeparator: () => ({}),
  dropdownIndicator: provided => ({
    ...provided,
    color: '#282d33',
    paddingBottom: 6,
    paddingTop: 6
  }),
  menu: provided => ({
    ...provided,
    borderRadius: 'none',
    boxShadow:
      '0 0.0625rem 0.0625rem 0rem rgba(3,3,3,.05),0 0.25rem 0.5rem 0.25rem rgba(3,3,3,.06)',
    paddingBottom: '0.5rem',
    paddingTop: '0.5rem',
    marginTop: 4,
    width: 'calc(100% - 4px)',
    marginRight: 2,
    marginLeft: 2
  }),
  option: (provided, { isFocused }) => ({
    ...provided,
    cursor: 'pointer',
    textAlign: 'left',
    color: '#282d33',
    fontSize: '1rem',
    fontWeight: '400',
    lineHeight: '1.5',
    paddingBottom: '0.5rem',
    paddingTop: '0.5rem',
    paddingLeft: '1rem',
    paddingRight: '1rem',
    backgroundColor: isFocused ? '#ededed' : 'transparent',
    '&:hover': {
      backgroundColor: '#ededed'
    }
  }),
  placeholder: provided => ({
    ...provided,
    left: 20
  }),
  singleValue: provided => ({
    ...provided,
    left: 20,
    maxWidth: 'calc(100% - 28px)'
  })
};

export default style;
