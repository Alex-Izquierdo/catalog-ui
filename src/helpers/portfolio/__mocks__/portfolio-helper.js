export function fetchPortfolioByName(name = '') {
  return new Promise((res) => {
    if (name === 'should-conflict') {
      return res({
        data: [{ name: 'should-conflict', id: 'conflict-id' }]
      });
    }

    return res({
      data: [],
      meta: { count: 0 }
    });
  });
}
