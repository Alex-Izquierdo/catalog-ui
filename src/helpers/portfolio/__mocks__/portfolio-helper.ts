import { ApiCollectionResponse, AnyObject } from '../../../types/common-types';

export const fetchPortfolioByName = (
  name = ''
): Promise<Partial<ApiCollectionResponse<AnyObject>>> => {
  return new Promise((res) => {
    if (name === 'should-conflict') {
      return res({
        data: [{ name: 'should-conflict', id: 'conflict-id' }]
      });
    }

    return res({
      data: [],
      meta: { count: 0, limit: 50, offset: 0 }
    });
  });
};
