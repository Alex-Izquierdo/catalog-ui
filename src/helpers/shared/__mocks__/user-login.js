import MockAdapter from 'axios-mock-adapter';
import { getAxiosInstance, getGraphqlInstance } from '../user-login';

export const mockApi = new MockAdapter(getAxiosInstance());
export const mockGraphql = new MockAdapter(getGraphqlInstance());
