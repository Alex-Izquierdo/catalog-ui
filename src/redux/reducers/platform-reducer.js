import {
  FETCH_PLATFORM,
  FETCH_PLATFORMS,
  FETCH_PLATFORM_ITEMS,
  FETCH_PLATFORM_ITEM,
  FILTER_PLATFORM_ITEMS,
  FETCH_MULTIPLE_PLATFORM_ITEMS,
  SET_LOADING_STATE,
  FETCH_PLATFORM_INVENTORIES,
  SET_INVENTORIES_LOADING_STATE
} from '../action-types';

// Initial State
export const platformInitialState = {
  isPlatformDataLoading: false,
  selectedPlatform: {},
  platforms: [],
  platformItems: {},
  platformInventories: [],
  platformItem: {},
  platform: {},
  filterValue: ''
};

// rename isPlatformLoading.. to isLoading so we can use common action for loading states

const setLoadingState = (state, { payload = true }) => ({
  ...state,
  isPlatformDataLoading: payload
});
const setPlatforms = (state, { payload }) => ({
  ...state,
  platforms: payload,
  isPlatformDataLoading: false
});
const setPlatformItems = (state, { payload, meta: { platformId } }) => ({
  ...state,
  platformItems: { ...state.platformItems, [platformId]: payload },
  isPlatformDataLoading: false
});
const setMultiplePlatformItems = (state, { payload }) => ({
  ...state,
  platformItems: { ...state.platformItems, ...payload },
  isPlatformDataLoading: false
});
const setPortfolioItems = (state, { payload }) => ({
  ...state,
  portfolioItem: payload,
  isPlatformDataLoading: false
});
const selectPlatform = (state, { payload }) => ({
  ...state,
  selectedPlatform: payload,
  isLoading: false
});
const filterPlatformItems = (state, { payload }) => ({
  ...state,
  filterValue: payload
});
const setInventoriesDataLoadingState = (state, { payload = true }) => ({
  ...state,
  isInventoriesDataLoading: payload
});
const setPlatformInventories = (state, { payload }) => ({
  ...state,
  platformInventories: { ...state.platformInventories, ...payload },
  isInventoriesDataLoading: false
});

export default {
  [`${FETCH_PLATFORMS}_PENDING`]: setLoadingState,
  [`${FETCH_PLATFORMS}_FULFILLED`]: setPlatforms,
  [`${FETCH_PLATFORM_ITEMS}_PENDING`]: setLoadingState,
  [`${FETCH_PLATFORM_ITEMS}_FULFILLED`]: setPlatformItems,
  [`${FETCH_PLATFORM_ITEM}_PENDING`]: setLoadingState,
  [`${FETCH_PLATFORM_ITEM}_FULFILLED`]: setPortfolioItems,
  [`${FETCH_PLATFORM}_PENDING`]: setLoadingState,
  [`${FETCH_PLATFORM}_FULFILLED`]: selectPlatform,
  [`${FILTER_PLATFORM_ITEMS}_FULFILLED`]: filterPlatformItems,
  [`${FETCH_MULTIPLE_PLATFORM_ITEMS}_FULFILLED`]: setMultiplePlatformItems,
  [SET_LOADING_STATE]: setLoadingState,
  [`${FETCH_PLATFORM_INVENTORIES}_PENDING`]: setInventoriesDataLoadingState,
  [`${FETCH_PLATFORM_INVENTORIES}_FULFILLED`]: setPlatformInventories,
  [SET_INVENTORIES_LOADING_STATE]: setInventoriesDataLoadingState
};
