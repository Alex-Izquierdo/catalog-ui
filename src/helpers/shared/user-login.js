import axios from 'axios';
import { WorkflowApi } from '@redhat-cloud-services/approval-client';
import { AdminsApi as CatalogAdminsApi } from '@redhat-cloud-services/catalog-client';
import { DefaultApi, ApiClient as TopologicalInventoryApiClient } from '@manageiq/topological_inventory';

import { TOPOLOGICAL_INVENTORY_API_BASE, CATALOG_API_BASE, APPROVAL_API_BASE, RBAC_API_BASE } from '../../utilities/constants';
import { AccessApi, PrincipalApi, GroupApi, ApiClient } from 'rbac_api_jsclient';

const axiosInstance = axios.create();

const resolveInterceptor = response => response.data || response;

axiosInstance.interceptors.response.use(resolveInterceptor);

const catalogAdmin = new CatalogAdminsApi(undefined, CATALOG_API_BASE, axiosInstance);
const workflowApi = new WorkflowApi(undefined, APPROVAL_API_BASE, axiosInstance);

const defaultClient = TopologicalInventoryApiClient.instance;
defaultClient.basePath = TOPOLOGICAL_INVENTORY_API_BASE;

let userTopologicalApi = new DefaultApi();

export function getTopologicalUserApi() {
  return userTopologicalApi;
}

export function getUserApi() {
  return catalogAdmin;
}

const defaultRbacClient = ApiClient.instance;
defaultRbacClient.basePath = RBAC_API_BASE;

let rbacAccessApi = new AccessApi();
let rbacPrincipalApi = new PrincipalApi();
let rbacGroupApi = new GroupApi();

export function getRbacAccessApi() {
  return rbacAccessApi;
}

export function getRbacPrincipalApi() {
  return rbacPrincipalApi;
}

export function getRbacGroupApi() {
  return rbacGroupApi;
}

export function getWorkflowApi() {
  return workflowApi;
}

export async function getUserAccess() {
  let application = 'rbac'; // String | The application name to obtain access for the principal
  let opts = {
    pageSize: 10, // Number | Parameter for selecting the amount of data in a page.
    page: 1 // Number | Parameter for selecting the page of data.
  };

  rbacAccessApi.getPrincipalAccess(application, opts).then((data) => {
    console.log('User Access API called successfully. Returned data: ' + data);
  }, (error) => {
    console.error(error);
  });
}
