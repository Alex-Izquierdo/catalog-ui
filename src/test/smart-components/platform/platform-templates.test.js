import React from 'react';
import configureStore from 'redux-mock-store' ;
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import promiseMiddleware from 'redux-promise-middleware';
import { Route, MemoryRouter } from 'react-router-dom';
import { notificationsMiddleware } from '@redhat-cloud-services/frontend-components-notifications/';
import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import PlatformTemplates from '../../../smart-components/platform/platform-templates';
import { TOPOLOGICAL_INVENTORY_API_BASE, SOURCES_API_BASE } from '../../../utilities/constants';
import { platformInitialState } from '../../../redux/reducers/platform-reducer';
import { FETCH_PLATFORM, FETCH_PLATFORM_ITEMS } from '../../../redux/action-types';
import { act } from 'react-dom/test-utils';
import { mockApi } from '../../__mocks__/user-login';

describe('<PlatformTemplates />', () => {
  let initialProps;
  const middlewares = [ thunk, promiseMiddleware, notificationsMiddleware() ];
  let mockStore;
  let initialState;

  const ComponentWrapper = ({ store, initialEntries = [ '/foo' ], children }) => (
    <Provider store={ store }>
      <MemoryRouter initialEntries={ initialEntries }>
        <Route path='/platforms/detail/:id/platform-templates'>
          { children }
        </Route>
      </MemoryRouter>
    </Provider>
  );

  beforeEach(() => {
    initialProps = {};
    mockStore = configureStore(middlewares);
    initialState = {
      platformReducer: {
        ...platformInitialState,
        selectedPlatform: {
          id: '1',
          name: 'Foo'
        },
        platformItems: {
          1: {
            meta: {
              limit: 50,
              offset: 0,
              count: 0
            }
          }
        }
      }
    };
  });

  it('should render correctly', () => {
    const store = mockStore(initialState);
    const wrapper = shallow(
      <ComponentWrapper store={ store } initialEntries={ [ '/platforms/detail/1/platform-templates' ] }>
        <PlatformTemplates store={ mockStore(initialState) } { ...initialProps } />);
      </ComponentWrapper>).find(PlatformTemplates);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should mount and fetch data', async done => {
    mockApi.onGet(`${SOURCES_API_BASE}/sources/1`).replyOnce(200, { name: 'Foo' });
    mockApi.onGet(`${TOPOLOGICAL_INVENTORY_API_BASE}/sources/1/service_offerings?filter[archived_at][nil]&limit=50&offset=0`)
    .replyOnce(200, {
      data: [{
        id: '1',
        name: 'Offering 1'
      }]
    });

    const expectedActions = [
      { type: `${FETCH_PLATFORM}_PENDING` },
      { type: `${FETCH_PLATFORM_ITEMS}_PENDING`,
        meta: { platformId: '1' }
      },
      { type: `${FETCH_PLATFORM}_FULFILLED`, payload: { name: 'Foo' }},
      {
        meta: { platformId: '1' },
        type: `${FETCH_PLATFORM_ITEMS}_FULFILLED`,
        payload: { data: [{ id: '1', name: 'Offering 1' }]}
      }];

    const store = mockStore(initialState);
    await act(async () => {
      mount(<ComponentWrapper store={ store } initialEntries={ [ '/platforms/detail/1/platform-templates' ] }>
        <PlatformTemplates { ...initialProps } store={ mockStore(initialState) }/>
      </ComponentWrapper>);
    });
    setImmediate(() => {
      expect(store.getActions()).toEqual(expectedActions);
      done();
    });
  });
});
