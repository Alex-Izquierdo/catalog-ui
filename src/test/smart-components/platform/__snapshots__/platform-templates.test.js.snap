// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlatformTemplates /> should render correctly 1`] = `
<PlatformTemplates
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
